pkg/controllermanager/controller/event/reconciler_test.go

```diff
@@ -0,0 +1,154 @@
+			ObjectMeta:    metav1.ObjectMeta{Name: eventWithoutInvolvedObjectName},
+			LastTimestamp: metav1.Time{Time: time.Date(2022, 0, 0, 0, 0, 0, 0, time.UTC)},
+		}
+		nonGardenerAPIGroupEvent = &corev1.Event{
+			ObjectMeta:     metav1.ObjectMeta{Name: nonGardenerAPIGroupEventName},
+			LastTimestamp:  metav1.Time{Time: time.Date(2022, 0, 0, 0, 0, 0, 0, time.UTC)},
+			InvolvedObject: corev1.ObjectReference{Kind: "Shoot", APIVersion: "v1"},
+		}
+
+		cfg = config.EventControllerConfiguration{
+			TTLNonShootEvents: ttl,
+		}
+
+		reconciler = &Reconciler{Client: fakeClient, Config: cfg}
+	})
+
+	It("should return nil because object not found", func() {
+		Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(nonShootEvent), &corev1.Event{})).To(BeNotFoundError())
+
+		result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: nonShootEventName}})
+		Expect(result).To(Equal(reconcile.Result{}))
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	Context("shoot events", func() {
+		It("should ignore them", func() {
+			Expect(fakeClient.Create(ctx, shootEvent)).To(Succeed())
+			Expect(fakeClient.Get(ctx, client.ObjectKeyFromObject(shootEvent), &corev1.Event{})).To(Succeed())
+
+			result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: shootEventName}})
+			Expect(result).To(Equal(reconcile.Result{}))
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("non-shoot events", func() {
+		Context("ttl is not yet reached", func() {
+			BeforeEach(func() {
+				nowFunc = func() time.Time {
+					return time.Date(2022, 0, 0, 0, 0, 0, 0, time.UTC)
+				}
+			})
+
+			It("should requeue non-shoot events", func() {
+				Expect(fakeClient.Create(ctx, nonShootEvent)).To(Succeed())
+
+				result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: nonShootEventName}})
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: ttl.Duration}))
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should requeue events with an empty involvedObject", func() {
+				Expect(fakeClient.Create(ctx, eventWithoutInvolvedObject)).To(Succeed())
+
+				result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: eventWithoutInvolvedObjectName}})
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: ttl.Duration}))
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("should requeue events with non Gardener APIGroup", func() {
+				Expect(fakeClient.Create(ctx, nonGardenerAPIGroupEvent)).To(Succeed())
+				result, err := reconciler.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: nonGardenerAPIGroupEventName}})
+				Expect(result).To(Equal(reconcile.Result{RequeueAfter: ttl.Duration}))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("ttl is reached", func() {
+			BeforeEach(func() {
+				nowFunc = func() time.Time {
```

Comment:
`nowFunc` needs to be reset.
Ideally, you would avoid the global var for mocking `time.Now` and use a clock instead.